src/proof/arith_proof.cpp

```diff
@@ -0,0 +1,982 @@
+        // n1 is an equality/iff, but n2 is a predicate
+        if(n1[0] == n2) {
+          n1 = n1[1];
+          ss << "(symm _ _ _ " << ss1.str() << ") (pred_eq_t _ " << ss2.str() << ")";
+        } else if(n1[1] == n2) {
+          n1 = n1[0];
+          ss << ss1.str() << " (pred_eq_t _ " << ss2.str() << ")";
+        } else {
+          Unreachable();
+        }
+      } else if(n2.getKind() == kind::EQUAL || n2.getKind() == kind::IFF) {
+        // n2 is an equality/iff, but n1 is a predicate
+        if(n2[0] == n1) {
+          n1 = n2[1];
+          ss << "(symm _ _ _ " << ss2.str() << ") (pred_eq_t _ " << ss1.str() << ")";
+        } else if(n2[1] == n1) {
+          n1 = n2[0];
+          ss << ss2.str() << " (pred_eq_t _ " << ss1.str() << ")";
+        } else {
+          Unreachable();
+        }
+      } else {
+        // Both n1 and n2 are prediacates. Don't know what to do...
+        Unreachable();
+      }
+
+      ss << ")";
+    }
+    out << ss.str();
+    Debug("pf::arith") << "\n++ trans proof done, have proven " << n1 << std::endl;
+    return n1;
+  }
+
+  case eq::MERGED_ARRAYS_ROW: {
+    Debug("pf::arith") << "eq::MERGED_ARRAYS_ROW encountered in Arith_PROOF" << std::endl;
+    Unreachable();
+
+    Debug("pf::arith") << "row lemma: " << pf->d_node << std::endl;
+    Assert(pf->d_node.getKind() == kind::EQUAL);
+    TNode t1, t2, t3, t4;
+    Node ret;
+    if(pf->d_node[1].getKind() == kind::SELECT &&
+       pf->d_node[1][0].getKind() == kind::STORE &&
+       pf->d_node[0].getKind() == kind::SELECT &&
+       pf->d_node[0][0] == pf->d_node[1][0][0] &&
+       pf->d_node[0][1] == pf->d_node[1][1]) {
+      t2 = pf->d_node[1][0][1];
+      t3 = pf->d_node[1][1];
+      t1 = pf->d_node[0][0];
+      t4 = pf->d_node[1][0][2];
+      ret = pf->d_node[1].eqNode(pf->d_node[0]);
+      Debug("pf::arith") << "t1 " << t1 << "\nt2 " << t2 << "\nt3 " << t3 << "\nt4 " << t4 << "\n";
+    } else {
+      Assert(pf->d_node[0].getKind() == kind::SELECT &&
+             pf->d_node[0][0].getKind() == kind::STORE &&
+             pf->d_node[1].getKind() == kind::SELECT &&
+             pf->d_node[1][0] == pf->d_node[0][0][0] &&
+             pf->d_node[1][1] == pf->d_node[0][1]);
+      t2 = pf->d_node[0][0][1];
+      t3 = pf->d_node[0][1];
+      t1 = pf->d_node[1][0];
+      t4 = pf->d_node[0][0][2];
+      ret = pf->d_node;
+      Debug("pf::arith") << "t1 " << t1 << "\nt2 " << t2 << "\nt3 " << t3 << "\nt4 " << t4 << "\n";
+    }
+    out << "(row _ _ ";
+    tp->printTerm(t2.toExpr(), out, map);
+    out << " ";
+    tp->printTerm(t3.toExpr(), out, map);
+    out << " ";
+    tp->printTerm(t1.toExpr(), out, map);
+    out << " ";
+    tp->printTerm(t4.toExpr(), out, map);
+    out << " " << ProofManager::getLitName(t2.eqNode(t3)) << ")";
+    return ret;
+  }
+
+  case eq::MERGED_ARRAYS_ROW1: {
+    Debug("pf::arith") << "eq::MERGED_ARRAYS_ROW1 encountered in Arith_PROOF" << std::endl;
+    Unreachable();
+
```

Comment:
Dead code? Maybe add a comment to explain this and comment out the unreachable things?